packages/tasit-action/src/TasitAction.js

```diff
@@ -20,76 +20,85 @@ class Utils {
 }
 
 class Subscription {
-  #emitter;
-  #events = [];
+  #ethersEventEmitter;
+  #events = new Map();
 
   constructor(eventEmitter) {
-    this.#emitter = eventEmitter;
+    this.#ethersEventEmitter = eventEmitter;
   }
 
-  removeListener = (eventName, listener) => {
-    this.#events = this.#events.filter(event => {
-      if (event.eventName === eventName && event.listener === listener) {
-        this.#emitter.removeListener(
-          event.wrappedEventName,
-          event.wrappedListener
-        );
-        return false;
-      }
-      return true;
-    });
+  _toEthersEventName = eventName => {
+    if (eventName === "confirmation") return "block";
+    return eventName;
   };
 
-  off = (eventName, listener) => {
-    this.removeListener(eventName, listener);
-  };
+  off = eventName => {
+    const event = this.#events.get(eventName);
+
+    if (!event) return;
 
-  listenerCount = eventName => {
-    return this.#events.filter(event => {
-      return event.eventName === eventName;
-    }).length;
+    if (eventName !== "error") {
+      const { listener } = event;
+
+      // Note: ethers.removeAllListeners(name) not working!
+      this.#ethersEventEmitter.removeListener(
+        this._toEthersEventName(eventName),
+        listener
```

Comment:
ethers.js EventEmitter has these methods: removeListener(name, listener) to remove one listener and removeAllListeners([name]) to remove all listeners of all events or all listeners of one event.
I don't know why, but removeAllListeners(name) aren't working properly. Should I investigate that error?

Reply:
I think `removeAllListeners(name)` wouldn't make sense for us since we differ from ethers.js in only having one listener per event.

Also we can remove `listener` as a param of `removeListener(name, listener)` for the same reason

`removeAllListeners()` still makes sense though, and that functionality would mostly mirror the ethers.js one since I think the `[name]` param was optional in ethers.js?

Reply:
Do you mean removeAllListeners(name) isn't working properly in ethers.js or in `tasit-action`?

Reply:
Sorry, I'll be redundant here because seems that I haven't understood you well.

A) `ethersEventEmitter` is `ethers.contract` or `ethers.provider` object. 
B) I've tried to use their `ethers.removeAllEventListeners()` but it didn't work properly, because of that I've used `ethers.removeEventListener(eventName, listener)` instead.
C) Our API has only the `off(eventName)` function to remove/unregister an event.

Reply:
https://github.com/ethers-io/ethers.js/issues/391

Reply:
PR https://github.com/tasitlabs/TasitSDK/pull/115

Reply:
Thanks for creating the issue on the ethers.js repo and the PR with the test case here. Resolving this!